simulator/src/main/java/peersim/kademlia/KademliaObserver.java

```diff
@@ -92,27 +110,45 @@ private static void writeMap(Map<String, Map<String, Object>> map, String filena
     }
   }
 
-  /** Write map find operation */
+  /**
+   * Writes a map of find operations to a file.
+   *
+   * @param map the map to write
+   * @param filename the name of the file to write to
+   */
   private static void writeMapFind(Map<String, Map<String, Object>> map, String filename) {
     try (FileWriter writer = new FileWriter(filename)) {
-      Set<String> keySet = map.entrySet().iterator().next().getValue().keySet();
-      String header = "";
-      for (Object key : keySet) {
-        header += key + ",";
-      }
+      // Define the expected order of keys in the header
+      String[] keys = {"src", "start", "stop", "messages", "hops", "id", "type"};

```

Comment:
Same here - let's read the keys from the map itself

Reply:
Done